kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import New from 'components/Topics/New/New';
+import { Route, Router } from 'react-router';
+import configureStore from 'redux-mock-store';
+import { RootState } from 'redux/interfaces';
+import { Provider } from 'react-redux';
+import { screen, waitFor } from '@testing-library/react';
+import { createMemoryHistory } from 'history';
+import fetchMock from 'fetch-mock-jest';
+import { clusterTopicNewPath, clusterTopicPath } from 'lib/paths';
+import userEvent from '@testing-library/user-event';
+import { render } from 'lib/testHelpers';
+
+import { createTopicPayload, createTopicResponsePayload } from './fixtures';
+
+const mockStore = configureStore();
+
+const clusterName = 'local';
+const topicName = 'test-topic';
+
+const initialState: Partial<RootState> = {};
+const storeMock = mockStore(initialState);
+const historyMock = createMemoryHistory();
+const createTopicAPIPath = `/api/clusters/${clusterName}/topics`;
+
+const renderComponent = (history = historyMock, store = storeMock) =>
+  render(
+    <Router history={history}>
+      <Route path={clusterTopicNewPath(':clusterName')}>
+        <Provider store={store}>
+          <New />
+        </Provider>
+      </Route>
+      <Route path={clusterTopicPath(':clusterName', ':topicName')}>
+        New topic path
+      </Route>
+    </Router>
+  );
+
+describe('New', () => {
+  beforeEach(() => {
+    fetchMock.reset();
+  });
+
+  it('validates form', async () => {
+    const mockedHistory = createMemoryHistory({
+      initialEntries: [clusterTopicNewPath(clusterName)],
+    });
+    jest.spyOn(mockedHistory, 'push');
+    renderComponent(mockedHistory);
+
+    await waitFor(() => {
+      userEvent.click(screen.getByText('Send'));
+    });
+    await waitFor(() => {
+      expect(screen.getByText('name is a required field')).toBeInTheDocument();
+    });
+    await waitFor(() => {
+      expect(mockedHistory.push).toBeCalledTimes(0);
+    });
+  });
+
+  it('submits valid form', async () => {
+    const createTopicAPIPathMock = fetchMock.postOnce(
+      createTopicAPIPath,
+      createTopicResponsePayload,
+      {
+        body: createTopicPayload,
+      }
+    );
+    const mockedHistory = createMemoryHistory({
+      initialEntries: [clusterTopicNewPath(clusterName)],
+    });
+    jest.spyOn(mockedHistory, 'push');
+    renderComponent(mockedHistory);
+
+    await waitFor(() => {
+      userEvent.type(screen.getByPlaceholderText('Topic Name'), topicName);
+      userEvent.click(screen.getByText('Send'));
+    });
+
+    await waitFor(() =>
+      expect(mockedHistory.location.pathname).toBe(
+        clusterTopicPath(clusterName, topicName)
+      )
+    );
+    await waitFor(() => expect(mockedHistory.push).toBeCalledTimes(1));
```

Comment:
I'm not sure we need waitFor here. You have been waiting this on a previous line

Reply:
Y, it seems after we done waiting for `pathname` change all other mocks are already changed and we don't need to wait for them
thanks
![image](https://user-images.githubusercontent.com/12596177/151979076-17e72273-cc21-4a2a-b6e7-8c24f9c2cc71.png)